src/Elastic.Apm.AspNetCore/ApmMiddleware.cs

```diff
@@ -88,6 +86,22 @@ public async Task InvokeAsync(HttpContext context)
 					Headers = responseHeaders
 				};
 
+				if (context.User?.Identity != null && context.User.Identity.IsAuthenticated && context.User.Identity != null
+					&& transaction.Context.User == null)
+				{
+					transaction.Context.User = new User { UserName = context.User.Identity.Name };
+
+					var idQuery = context.User.Claims.Where(n => n.Type == "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier");

```

Comment:
would be nice if this could be configurable to other claim type or at least also check OAuth/OpenID claims in this case `sub`

Reply:
Great point, thanks! 👍  I think an extra config would be an overkill, so I added a fallback and we also look for OpenID claims.